inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -641,4 +631,20 @@ protected function get_url( $post ) {
 
 		return $url;
 	}
+
+	/**
+	 * Retrieve posts that should be excluded from the XML sitemap.
+	 *
+	 * @return array Array of post IDs to exclude.
+	 */
+	private function get_excluded_posts() {
+		/**
+		 * Filter posts out of the XML sitemap.
+		 *
+		 * @param array $excluded_posts An array of post IDs to exclude.
+		 */
+		$excluded_posts = apply_filters( 'wpseo_sitemap_excluded_posts', array() );
```

Comment:
I think we can close #8763 in favor of this? Maybe some of the code from that pull can be reused?